packages/web3-eth-accounts/src/index.js

```diff
@@ -351,13 +351,13 @@ Accounts.prototype.encrypt = function (privateKey, password, options) {
     if (kdf === 'pbkdf2') {
         kdfparams.c = options.c || 262144;
         kdfparams.prf = 'hmac-sha256';
-        derivedKey = cryp.pbkdf2Sync(Buffer.from(password), salt, kdfparams.c, kdfparams.dklen, 'sha256');
+        derivedKey = cryp.pbkdf2Sync(Buffer.from(password), Buffer.from(kdfparams.salt, 'hex'), kdfparams.c, kdfparams.dklen, 'sha256');
```

Comment:
Quite confusing that we are not using `salt` any more but `kdfparams.salt` all of a sudden. When I read the code above it should be the same, but still confusing.

Reply:
See above:

https://github.com/michaelsbradleyjr/web3.js/blob/d7adf702f815537376e7b8adb8fbdacfece17435/packages/web3-eth-accounts/src/index.js#L348

```
var kdfparams = {
    dklen: options.dklen || 32,
    salt: salt.toString('hex')
};
```

That code was already there and normalized the `salt` whether it was a string or buffer. The problem was we weren't using the normalized value and it resulted in encrypted output that couldn't be decrypted. `ethereumjs-wallet` has the same bug and I made a PR to fix it (still open).